### PR TITLE
db/driver: fix MySQL driver compilation warnings

### PR DESCRIPTION
**Pull request by nilason**
Addresses a few compilation warnings with MySQL driver and in addition improves the configuration output.

(This does **not** add the driver to the CI tests; eg. PostgreSQL driver is also not included, thus that would be another issue.)

### CODE REVIEW

File: db/drivers/mysql/driver.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #include "globals.h"
 #include "proto.h"
 
-int db__driver_init(int argc, char *argv[])
+int db__driver_init(int argc UNUSED, char *argv[] UNUSED)
```
Comment (echoix) [inline db/drivers/mysql/driver.c:19]: I've never seen this yet, what does it mean? It's like an extra "word" (not as in a computer word) in a function signature?

I've tried looking it up, the closest I got was GNU C `__attribute__`. In that case, does it still compile correctly with clang?

Comment (nilason) [reply db/drivers/mysql/driver.c:19]: https://github.com/OSGeo/grass/blob/aaecfc8957ab906785b3231da2aed7d0861ce71a/include/grass/gis.h#L45

Comment (echoix) [reply db/drivers/mysql/driver.c:19]: Thanks! That's clear ;)

**Comment by nilason** [comment]
Auto-merge enabled, only takes an approval to merge this quite trivial fix...

**Review by echoix — APPROVED** [review]

**Pull request merged**